aws/resource_aws_rds_cluster.go

```diff
@@ -650,6 +697,117 @@ func resourceAwsRDSClusterCreate(d *schema.ResourceData, meta interface{}) error
 			return err
 		}
 
+	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok {
+		pointInTime := v.([]interface{})[0].(map[string]interface{})
+
+		restoreToTimeAttr := pointInTime["restore_to_time"].(string)
+		useLatestRestorableTimeAttr := pointInTime["use_latest_restorable_time"].(bool)
+
+		if useLatestRestorableTimeAttr && restoreToTimeAttr != "" {
+			return fmt.Errorf(`provider.aws: aws_rds_cluster: %s: Either "restore_to_time" or "use_latest_restorable_time" must be set`, d.Get("database_name").(string))
+		}
+
+		createOpts := &rds.RestoreDBClusterToPointInTimeInput{
+			DBClusterIdentifier:       aws.String(identifier),
+			DeletionProtection:        aws.Bool(d.Get("deletion_protection").(bool)),
+			SourceDBClusterIdentifier: aws.String(pointInTime["source_cluster_identifier"].(string)),
+			Tags:                      tags,
+		}
+
+		if attr, ok := pointInTime["restore_type"]; ok {
+			createOpts.RestoreType = aws.String(attr.(string))
+		}
+
+		if useLatestRestorableTimeAttr {
+			createOpts.UseLatestRestorableTime = aws.Bool(useLatestRestorableTimeAttr)
+		} else if restoreToTimeAttr != "" {
+			restoreToTime, _ := time.Parse(time.RFC3339, restoreToTimeAttr)
+			createOpts.RestoreToTime = aws.Time(restoreToTime)
+		} else {
```

Comment:
In the event neither `restore_to_time` nor `use_latest_restorable_time` is set, we hit this `else` condition so the error message here is unfortunately not applicable .. see comment above for proposed refactor